### DISCUSSION TITLE
Consider making BlobRef use interface instead of class

### DISCUSSION THREAD

**Discussion by mikuhl-dev** [bluesky-social/atproto#1146] [Protocol (atproto)]
[BlobRef](https://github.com/bluesky-social/atproto/blob/main/packages/lexicon/src/blob-refs.ts#L26) seems to be the only thing that uses a class, and this single object is preventing it from being serialized by [Qwik](https://qwik.builder.io/)

**Comment by devinivy** [comment]
The purpose of `BlobRef` is to help developers handle edge-cases around a legacy blob format (there's a legacy and a revised/final format for blobs), so I don't think this would be a trivial change.  I don't know exactly how you're using this with qwik, but perhaps you could use the "plain json" version of these values, which can be generated by passing them through `lexToJson` from the `@atproto/lexicon` package.
